lib/Conversion/TritonToLinearAlgebraSubprograms/TritonToLinearAlgebraSubprogramsPass.cpp

```diff
@@ -0,0 +1,173 @@
+         ptrType, intType, ptrType, intType, floatType,
+         ptrType, intType}, {});
+
+    bool usingF64 = floatType.isF64();
+    const char *funcName = usingF64 ? "cblas_dgemm" : "cblas_sgemm";
+    auto func = module.lookupSymbol<func::FuncOp>(funcName);
+    if (!func) {
+      OpBuilder::InsertionGuard guard(rewriter);
+      rewriter.setInsertionPointToStart(module.getBody());
+      func = rewriter.create<func::FuncOp>(loc, funcName, funcType);
+      func.setVisibility(SymbolTable::Visibility::Private);
+    }
+
+    auto memrefToPointer = [&rewriter, &loc, &int64Type, &ptrType](Value &memref) {
+      auto indexPtr = rewriter.create<memref::ExtractAlignedPointerAsIndexOp>(loc, memref);
+      auto castOp = rewriter.create<arith::IndexCastOp>(loc, int64Type, indexPtr);
+      return rewriter.create<LLVM::IntToPtrOp>(loc, ptrType, castOp);
+    };
+
+    auto tensorToPointer = [&rewriter, &loc, &memrefToPointer](Value &V, RankedTensorType &T) {
+      Value memref = rewriter.create<bufferization::ToMemrefOp>(loc, MemRefType::get(T.getShape(), T.getElementType()), V);
+      return memrefToPointer(memref);
+    };
+
+    Value ptrA = tensorToPointer(A, tensorA);
+    Value ptrB = tensorToPointer(B, tensorB);
+    Value ptrC = memrefToPointer(memrefC);
+
+    int32_t M = tensorA.getShape()[0];
+    int32_t K = tensorA.getShape()[1];
+    int32_t N = tensorB.getShape()[1];
+
+    Value alpha = rewriter.create<arith::ConstantOp>(loc, floatType, usingF64 ? rewriter.getF64FloatAttr(1.0) : rewriter.getF32FloatAttr(1.0));
+    Value beta = alpha;
+
+    auto constOp = [&rewriter, &loc, &intType](int32_t V) {
+      return rewriter.create<arith::ConstantOp>(loc, intType, rewriter.getI32IntegerAttr(V));
+      };
+    Value CblasRowMajor = constOp(101), CblasNoTrans = constOp(111);
+    Value MVal = constOp(M), NVal = constOp(N), KVal = constOp(K);
+    Value LDA = KVal, LDB = NVal, LDC = NVal;
+
+    auto funcOp = rewriter.create<func::CallOp>(loc, func, ValueRange{
+        CblasRowMajor, CblasNoTrans, CblasNoTrans,
+        MVal, NVal, KVal,
+        alpha, ptrA, LDA,
+        ptrB, LDB, beta,
+        ptrC, LDC
+    });
+
+    auto toTensorOp = rewriter.create<bufferization::ToTensorOp>(loc,
+        tensorC, memrefC, true /* restrict */, true /* writable */);
+    rewriter.replaceOp(op, toTensorOp);
+    return success();
+  }
+};
+
+class TritonToLinearAlgebraSubprogramsPass
+    : public triton::impl::TritonToLinearAlgebraSubprogramsBase<TritonToLinearAlgebraSubprogramsPass> {
+  using TritonToLinearAlgebraSubprogramsBase<
+      TritonToLinearAlgebraSubprogramsPass>::TritonToLinearAlgebraSubprogramsBase;
+
+public:
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry
+        .insert<linalg::LinalgDialect, func::FuncDialect, arith::ArithDialect, math::MathDialect,
+            affine::AffineDialect, scf::SCFDialect, tensor::TensorDialect, LLVM::LLVMDialect, triton::TritonDialect>();
+  }
```

Comment:
We're missing the BufferizationDialect here. Without it we cannot run this pass individually.

Reply:
thank you, fixed!